pkgs/applications/misc/googleearth-pro/default.nix

```diff
@@ -1,41 +1,54 @@
-{ lib, stdenv, fetchurl, glibc, libGLU, libGL, freetype, glib, libSM, libICE, libXi, libXv
-, libXrender, libXrandr, libXfixes, libXcursor, libXinerama, libXext, libX11, libXcomposite
-, libxcb, sqlite, zlib, fontconfig, dpkg, libproxy, libxml2, gst_all_1, dbus, makeWrapper }:
-
+{ lib
+, stdenv
+, mkDerivation
+, fetchurl
+, ffmpeg_3
+, freetype
+, gdal_2
+, glib
+, libGL
+, libGLU
+, libICE
+, libSM
+, libXi
+, libXv
+, libav_12
+
+, libXrender
+, libXrandr
+, libXfixes
+, libXcursor
+, libXinerama
+, libXext
+, libX11
+, libXcomposite
+
+, libxcb
+, sqlite
+, zlib
+, fontconfig
+, dpkg
+, libproxy
+, libxml2
+, gst_all_1
+, dbus
+, makeWrapper
+
+, qtlocation
+, qtwebkit
+, qtx11extras
+, qtsensors
+, qtscript
+
+, xkeyboardconfig
+, autoPatchelfHook
+}:
 let
   arch =
     if stdenv.hostPlatform.system == "x86_64-linux" then "amd64"
     else throw "Unsupported system ${stdenv.hostPlatform.system} ";
-  fullPath = lib.makeLibraryPath [
-    glibc
-    glib
-    stdenv.cc.cc
-    libSM
-    libICE
-    libXi
-    libXv
-    libGLU libGL
-    libXrender
-    libXrandr
-    libXfixes
-    libXcursor
-    libXinerama
-    libXcomposite
-    freetype
-    libXext
-    libX11
-    libxcb
-    sqlite
-    zlib
-    fontconfig
-    libproxy
-    libxml2
-    dbus
-    gst_all_1.gstreamer
-    gst_all_1.gst-plugins-base
-  ];
```

Comment:
Is this just not the canonical way of adding / pathing the right deps into the binary, or there any other downsides to using `lib.makeLibraryPath` ?

Reply:
I'm not an expert of canonical-derivation making but using the autoPatchelfhook was the easiest way to replace all the bundled dynamic libraries with nixpkgs' one. Now that the cleanup has been done, I could use back makeLibraryPath.

Reply:
Ah, now I understand. I am also quite a beginner with nix :) So if you don't mind i have another question about the patching when I do `ldd googleearth-bin | grep ssl` now i get something like:
```
libssl.so.1.1 => /nix/store/yc41q33h5xrw1zbyw5hp1y1ga0jk9hwd-openssl-1.1.1k/lib/libssl.so.1.1 (0x00007f0965903000)
libcrypto.so.1.1 => /nix/store/yc41q33h5xrw1zbyw5hp1y1ga0jk9hwd-openssl-1.1.1k/lib/libcrypto.so.1.1 (0x00007f0965616000)
libssl.so.1.0.0 => /nix/store/gxzjr9h9vl2bngswd2w8d540xgmh4y1r-googleearth-pro-7.3.3.7786/opt/google/earth/pro/libssl.so.1.0.0 (0x00007f09618ee000)
```
does this mean, it now also uses the current nixpkg openssl and we could try removing the insecure `libssl-1.0.0` from the bundle?

Reply:
Nope that's an issue. openssl 1.0.0 is old, unmaintained and has some security issues. Nixpkgs doesn't even have a derivation for it anymore. That's why its using googleearth's one, every dynamic libraries in /nix/store/.....-googleearth-pro/opt/google/earth/pro/ are from the deb file.

Thus the libssl.so.1.0.0 => /nix/store/gxzjr9h9vl2bngswd2w8d540xgmh4y1r-googleearth-pro-7.3.3.7786/**opt/google/earth/pro**/libssl.so.1.0.0 (0x00007f09618ee000)


However you're right on the two others. They're from nixpkgs.